Src/VimCore/Interpreter_Interpreter.fs

```diff
@@ -469,6 +469,10 @@ type VimInterpreter
     member x.RunCall (callInfo: CallInfo) = 
         _statusUtil.OnError (Resources.Interpreter_CallNotSupported callInfo.Name)
 
+    member x.RunCallCSharpScript (callInfo: CallInfo) = 
+        let args = CallCSharpScriptEventArgs(callInfo)
+        _vimBuffer.RaiseCallCSharpScript args

```

Comment:
I think the best way to approach this is through the host. Essentially have a method on `IVimHost` that is used to invoke the script file.